app/service/data_svc.py

```diff
@@ -69,9 +69,17 @@ def __init__(self, dao, utility_svc):
 
     async def create_ability(self, ability_id, tactic, technique, name, test, description, platform, cleanup=None,
                              payload=None, parser=None):
-        await self.dao.create('core_attack',
-                              dict(attack_id=technique['attack_id'], name=technique['name'], tactic=tactic))
         entry = await self.dao.get('core_attack', dict(attack_id=technique['attack_id']))
+        if len(entry) == 0:
```

Comment:
good overall change; let's extract this change (createAttack()) to a separate private function in this module (data_svc).

Reply:
Taken care of.